Sources/Testing/Testing.docc/testing-asynchronous-code.md

```diff
@@ -0,0 +1,59 @@
+# Testing asynchronous code
+
+<!--
+This source file is part of the Swift.org open source project
+
+Copyright (c) 2024 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+Validate whether your code causes expected events to happen.
+
+## Overview
+
+Swift Testing integrates with Swift concurrency, meaning that in many situations you can test asynchronous code using standard Swift features.
+Mark your test function as `async` and, in the function body, `await` any asynchronous interactions:
+
+```swift
+@Test func asynchronousCalculationYieldsExpectedValue() async {
+    #expect(await asynchronousCalculation(with: 10) == 12)
```

Comment:
We don't currently recommend placing effectful expressions directly in `#expect()` or `#require()` because we can't expand them correctly on failure. Better would be:

```swift
let result = await asynchronousCalculation(with: 10)
#expect(result == 12)
```